### PR TITLE
SG-30910 Support latest version of Sphinx

### PR DESCRIPTION
**Pull request by carlos-villavicencio-adsk**
- Remove duplicate elements that cause issues when rendering docs with newer versions of Sphinx
- Detected by https://github.com/shotgunsoftware/tk-toolchain/pull/52

### CODE REVIEW

File: docs/core.rst [hunk 1]
```diff
@@ -390,16 +390,6 @@ The following exceptions are raised by the Toolkit Core API classes:
     :inherited-members:
     :members:
 
-.. autoclass:: TankFileDoesNotExistError
-    :show-inheritance:
-    :inherited-members:
-    :members:
-
-.. autoclass:: TankInvalidInterpreterLocationError
```
Comment (julien-lang) [inline docs/core.rst:398]: What are these things about? Does removing them impact the content of the documentation?

**Review by julien-lang — COMMENTED** [review]
Looks good. Do you have a preview we could look at?

Also, do we need to synchronize this change across all repositories? Or is there a way to merge something without having to rush merging everything else?

**Review by shaynacohen — APPROVED** [review]
This should only impact the https://developer.shotgridsoftware.com/tk-core/ sphinx documentation hosted from the tk-core repo, so it won't have any effect on the developer portal documentation it seems!

File: docs/core.rst (hunk 1 above)
Comment (carlos-villavicencio-adsk) [reply docs/core.rst:398]: Yes, [here](https://developer.shotgridsoftware.com/tk-core/descriptor.html) you can see `TankInvalidAppStoreCredentialsError` repeated.

[Here](https://developer.shotgridsoftware.com/tk-core/core.html#exceptions) you can see `sgtk.TankUnreadableFileError` , `sgtk.TankFileDoesNotExistError`, repeated. In this same page, `sgtk.TankInvalidInterpreterLocationError` is being displayed when it's already defined in the descriptor page. 

I have generated the preview locally, let me know if you want me to share a compressed file with you internally.

**Review by julien-lang — APPROVED** [review]
LGTM!

**Pull request merged**